src/utilities/geometry/Intersection.cpp

```diff
@@ -1243,12 +1326,13 @@ Polygon3d PolygonFromBoostPolygon(const BoostPolygon& boostPolygon, Point3dVecto
     Point3dVector hole;
     for (unsigned i = 0; i < inner.size() - 1; ++i) {
       Point3d point3d(inner[i].x(), inner[i].y(), 0.0);
-      Point3d resultPoint = getCombinedPoint(point3d, allPoints, tol);
+      const Point3d resultPoint = getCombinedPoint(point3d, allPoints, tol);
       // don't keep repeated vertices
       if ((i > 0) && (hole.back() == resultPoint)) {
         continue;
       }
-      hole.push_back(point3d);
+      // TODO: shouldn't this be resultPoint instead?! Or maybe the repeated check should use point3d in which case resultPoint is pointless
+      hole.emplace_back(std::move(point3d));
```

Comment:
@ggartside here's someone I noticed while doing the refactor. I think there's something wrong here. Please advise

Reply:
@jmarrec Me too - I think resultPoint should be used as getCombinedPoint might have modified the original Point3d

Reply:
@jmarrec Yes I think this should use resultPoint as the original point has potentially been modified by getCombinedPoint so I've made the change